### PR TITLE
RT-633-Summary-Card:Summary Card is not removed for disabled layers

### PR DESCRIPTION
**Pull request by yashpmohod**
## Overview
Add the ngIf condition to check layer is enabled or not.

image

## How it was tested
Tested using run showcase app locally.

## Checklist

- [x] The title contains a short meaningful summary
- [x] I have added a link to this PR in the Jira issue
- [x] I have described how this was tested
- [x] I have included screen shots for changes that affect the user interface
- [x] I have updated unit tests
- [x] I have run unit tests locally
- [ ] I have updated documentation (including README)

### CODE REVIEW

File: projects/ngx-charts-on-fhir/src/lib/fhir-chart-summary/fhir-chart-summary/fhir-chart-summary.component.html [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-      *ngFor="let layer of layerManager.selectedLayers$ | async"
-      [color]="colorService.getColorGradient(layer)"
-      [title]="layer.name"
-      icon="info"
-      class="dataset-summary mat-elevation-z2"
-    >
-      <dynamic-table [data]="summarize(layer, range)"></dynamic-table>
-    </fhir-chart-summary-card>
+    <div class="summary-card" *ngFor="let layer of layerManager.selectedLayers$ | async">
+      <fhir-chart-summary-card  *ngIf="layer.enabled"
```
Comment (semicolin) [inline projects/ngx-charts-on-fhir/src/lib/fhir-chart-summary/fhir-chart-summary/fhir-chart-summary.component.html:4]: This leaves an empty div for each hidden layer. It's barely noticeable with one hidden layer, but with more layers it can become a large gap.
image

File: projects/ngx-charts-on-fhir/src/lib/fhir-chart-summary/fhir-chart-summary/fhir-chart-summary.component.spec.ts [hunk 2]
```diff
@@ -67,20 +67,20 @@ describe('FhirChartSummaryComponent', () => {
     const layer: ManagedDataLayer = { id: '1', name: 'layer', datasets: [], scale: { id: 'test' } };
     layerManager.selectedLayers$.next([layer, layer, layer]);
     fixture.detectChanges();
-    const cards = fixture.debugElement.queryAll(By.directive(MockFhirChartSummaryCardComponent));
+    const cards = fixture.debugElement.queryAll(By.css('.summary-card'));
```
Comment (semicolin) [inline projects/ngx-charts-on-fhir/src/lib/fhir-chart-summary/fhir-chart-summary/fhir-chart-summary.component.spec.ts:70]: Why did you change this? we still want to make the sure the cards are rendered and not just empty divs, right?
Also, please add a test for `it('should not render a card for disabled layers')`

File: projects/ngx-charts-on-fhir/src/lib/fhir-chart-summary/fhir-chart-summary/fhir-chart-summary.component.html (hunk 1 above)
Comment (semicolin) [reply projects/ngx-charts-on-fhir/src/lib/fhir-chart-summary/fhir-chart-summary/fhir-chart-summary.component.html:4]: You can fix this either by using `<ng-container>` instead of `<div>`
or by creating an observable like:
```
  enabledLayers$ = this.selectedLayers$.pipe(map((layers) => layers.filter((layer) => layer.enabled)));
```
which might be useful in other places too.

**Review by semicolin — CHANGES_REQUESTED** [review]

**Review by amolht — DISMISSED** [review]

File: projects/ngx-charts-on-fhir/src/lib/fhir-chart-summary/fhir-chart-summary/fhir-chart-summary.component.spec.ts [hunk 3]
```diff
@@ -65,25 +66,36 @@ describe('FhirChartSummaryComponent', () => {
...
     const statistics: DebugElement = fixture.debugElement.query(By.directive(MockDynamicTableComponent));
     expect(statistics.componentInstance.data).toEqual([{ name: 'summary' }]);
   });
+  it('should not render a card for disabled layers', () => {
+    const enabledLayer: ManagedDataLayer = { id: '1', name: 'layer', datasets: [], scale: { id: 'test' },enabled:true };
+    const disabledLayer: ManagedDataLayer = { id: '1', name: 'layer', datasets: [], scale: { id: 'test' },enabled:false };
+    layerManager.selectedLayers$.next([enabledLayer, disabledLayer]);
+    layerManager.selectedLayers$.pipe(map((layers) => layers.filter((layer) => layer.enabled))).forEach(layer => {
+      layerManager.enabledLayers$.next(layer)
+    });
```
Comment (semicolin) [inline projects/ngx-charts-on-fhir/src/lib/fhir-chart-summary/fhir-chart-summary/fhir-chart-summary.component.spec.ts:96]: This is really confusing. The test should not be subscribing to a mock observable. This component no longer depends on selectedLayers$ so I don't understand why the test is mocking that.

It also should not be duplicating the enabledLayers$ logic from the real code. It seems like this test is really only testing itself.

Comment (semicolin) [reply projects/ngx-charts-on-fhir/src/lib/fhir-chart-summary/fhir-chart-summary/fhir-chart-summary.component.spec.ts:96]: Since you moved the logic for checking enabled layers into LayerManagerService, you should probably add an enabledLayers$ test in that class and delete this one.

File: projects/ngx-charts-on-fhir/src/lib/fhir-chart-summary/fhir-chart-summary/fhir-chart-summary.component.html [hunk 4]
```diff
@@ -1,13 +1,11 @@
...
-      <dynamic-table [data]="summarize(layer, range)"></dynamic-table>
-    </fhir-chart-summary-card>
+      <fhir-chart-summary-card *ngFor="let layer of layerManager.enabledLayers$ | async"
+        [color]="colorService.getColorGradient(layer)" 
+        [title]="layer.name" 
+        icon="info"
+        class="dataset-summary mat-elevation-z2">
+        <dynamic-table [data]="summarize(layer, range)"></dynamic-table>
+      </fhir-chart-summary-card>
   </ng-container>
```
Comment (semicolin) [inline projects/ngx-charts-on-fhir/src/lib/fhir-chart-summary/fhir-chart-summary/fhir-chart-summary.component.html:10]: please format

**Review by semicolin — CHANGES_REQUESTED** [review]

**Review by semicolin — APPROVED** [review]
